### PR TITLE
feat(AWTWA): add presence

### PR DESCRIPTION
**Pull request by emberantz**
## Description 
Hello, I added support for french anime streaming website AWTWA.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
image
image
image
image
</details>

### CODE REVIEW

File: websites/A/AWTWA/metadata.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"description": {
+		"en": "Discover the best free anime streaming site in high resolution.",
+		"fr": "Découvrez le meilleur site de streaming d'anime gratuit en haute résolution."
+	},
+	"url": [
+		"awtwa.site",
+		"www.awtwa.site"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/AoJisOX.png",
```
Comment (DarkVillager) [inline websites/A/AWTWA/metadata.json:17]: Logo's must be 512x512 as per [our guideliness](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/A/AWTWA/presence.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			presenceData.smallImageText = presenceStrings.watchOn;
+			presenceData.largeImageKey = Assets.Logo;
+			presenceData.smallImageKey = Assets.Play;
+		} else {
+			presenceData.details =
+				presenceStrings.onPage + document.querySelector("h1")?.textContent;
+			presenceData.state = `Saison ${
+				(document.querySelector("#season") as HTMLSelectElement).value
+			} - Épisode ${
+				(document.querySelector("#episode") as HTMLSelectElement).value
```
[the remark this answers is not in the document]
Comment (emberantz) [reply websites/A/AWTWA/presence.ts]: fixed.

File: websites/A/AWTWA/presence.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {},
+		{ pathname, href } = document.location;
+
+	if (pathname === "/") {
+		presenceData.details = presenceStrings.home;
+		presenceData.state = presenceStrings.branding;
+		presenceData.smallImageText = presenceStrings.websiteName;
+		presenceData.largeImageKey = Assets.Logo;
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:40]:
```suggestion
```

File: websites/A/AWTWA/presence.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+presence.on(
+	"iFrameData",
+	(data: { currentTime: number; duration: number; paused: boolean }) => {
+		iFrameData = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {},
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:33]:
```suggestion
	const presenceData: PresenceData = {
	largeImageKey: Assets.Logo
	},
```

File: websites/A/AWTWA/presence.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+		presenceData.details = presenceStrings.home;
+		presenceData.state = presenceStrings.branding;
+		presenceData.smallImageText = presenceStrings.websiteName;
+		presenceData.largeImageKey = Assets.Logo;
+		presenceData.smallImageKey = Assets.Viewing;
+	} else if (pathname === "/catalog") {
+		presenceData.details = presenceStrings.browsing;
+		presenceData.state = presenceStrings.branding;
+		presenceData.smallImageText = presenceStrings.websiteName;
+		presenceData.largeImageKey = Assets.Logo;
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:46]:
```suggestion
```

File: websites/A/AWTWA/presence.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+				.split("url(")[1]
+				.split(")")[0]
+				.replace(/['"]/g, ""),
+			isPrivacyMode = await presence.getSetting("privacy");
+
+		if (isPrivacyMode) {
+			presenceData.details = `${presenceStrings.onPage}un anime`;
+			presenceData.state = presenceStrings.branding;
+			presenceData.smallImageText = presenceStrings.watchOn;
+			presenceData.largeImageKey = Assets.Logo;
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:62]:
```suggestion
```

File: websites/A/AWTWA/presence.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+		} else {
+			presenceData.details =
+				presenceStrings.onPage + document.querySelector("h1")?.textContent;
+			presenceData.state = `Saison ${
+				document.querySelector<HTMLSelectElement>("#season").value
+			} - Épisode ${
+				document.querySelector<HTMLSelectElement>("#episode").value
+			}`;
+			presenceData.smallImageText = presenceStrings.watchOn;
+			presenceData.largeImageKey = poster;
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:73]:
```suggestion
			presenceData.largeImageKey = poster ?? Assets.Logo;
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/A/AWTWA/presence.ts [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+	Logo = "https://i.imgur.com/n8kQ5rV.png",
+}
+
+let iFrameData = {
+	currentTime: -1,
+	duration: -1,
+	paused: true,
+};
+
+const animeBrowsingTimestamp = Math.floor(Date.now() / 1000),
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:15]:
```suggestion
const browsingTimestamp = Math.floor(Date.now() / 1000),
```

File: websites/A/AWTWA/presence.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+		presenceData.smallImageText = presenceStrings.websiteName;
+		presenceData.largeImageKey = Assets.Logo;
+		presenceData.smallImageKey = Assets.Viewing;
+	} else if (pathname === "/catalog") {
+		presenceData.details = presenceStrings.browsing;
+		presenceData.state = presenceStrings.branding;
+		presenceData.smallImageText = presenceStrings.websiteName;
+		presenceData.largeImageKey = Assets.Logo;
+		presenceData.smallImageKey = Assets.Search;
+		presenceData.startTimestamp = animeBrowsingTimestamp;
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:48]:
```suggestion
		presenceData.startTimestamp = browsingTimestamp;
```

File: websites/A/AWTWA/presence.ts [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+					iFrameData.currentTime,
+					iFrameData.duration
+				);
+				presenceData.smallImageKey = Assets.Play;
+				presenceData.smallImageText = "En train de regarder";
+			} else {
+				presenceData.smallImageKey = Assets.Pause;
+				presenceData.smallImageText = "En pause";
+				delete presenceData.endTimestamp;
+			}
```
Comment (DarkVillager) [inline websites/A/AWTWA/presence.ts:87]:
```suggestion
			if (!iFrameData?.paused) {
				[, presenceData.endTimestamp] = presence.getTimestamps(
					iFrameData.currentTime,
					iFrameData.duration
				);
				}
				presenceData.smallImageKey = iFrameData?.paused ? Assets.Pause : Assets.Play;
				presenceData.smallImageText = iFrameData?.paused ? "En pause" : "En train de regarder";
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
If you have any questions about any of the requested changes feel free to ask in the discord server or in my dms on discord.

File: websites/A/AWTWA/metadata.json [hunk 11]
```diff
@@ -0,0 +1,40 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.10",
+	"author": {
+		"id": "815513219120103424",
+		"name": "ember"
```
Comment (DarkVillager) [inline websites/A/AWTWA/metadata.json:5]:
```suggestion
		"name": "emberantz"
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]
Must match discord name other than that LGTM

**Review by Bas950 — APPROVED** [review]

**Review by DarkVillager — APPROVED** [review]

**Review by Timeraa — APPROVED** [review]

**Pull request merged**